CRM/Streetimport/GP/Handler/TEDIContactRecordHandler.php

```diff
@@ -350,17 +353,51 @@ protected function getDate($record) {
    *
    * @param $contact_id
    * @param $record
+   * @param $parent_activity_id
    *
    * @throws \CiviCRM_API3_Exception
    */
-  public function performContactBaseUpdates($contact_id, $record) {
+  public function performContactBaseUpdates($contact_id, $record, $parent_activity_id) {
     //Contact Entity
     $contact_params = $this->getPreparedContactParams($record);
     $this->updateContact($contact_params, $contact_id, $record);
 
+    // Selection date
+    $selection_date = Api4\Activity::get(FALSE)
+      ->addSelect('created_date')
+      ->addWhere('id', '=', $parent_activity_id)
+      ->setLimit(1)
+      ->execute()
+      ->first()['created_date'];
+
     //Address Entity
-    $address_params = $this->getPreparedAddressParams($record);
-    $this->createOrUpdateAddress($contact_id, $address_params, $record);
+    if ($this->addressChangeRecordedSince($contact_id, $selection_date, $record)) {
+      $config = CRM_Streetimport_Config::singleton();
+      $address_attributes = $config->getAllAddressAttributes();
+
+      $current_address = Api4\Address::get(FALSE)
+        ->addSelect(...$address_attributes)
+        ->addWhere('contact_id', '=', $contact_id)
+        ->addWhere('is_primary', '=', TRUE)
+        ->setLimit(1)
+        ->execute()
+        ->first();
+
+      $this->createManualUpdateActivity(
+        $contact_id,
+        'Recent address change detected',
+        $record,
+        'activities/ManualAddressUpdate.tpl',
+        [
+          'title'    => 'Recent address change detected',
+          'subtitle' => "The primary address of contact #$contact_id has changed since the last call",
+          'fields'   => $address_attributes,
+          'address'  => $current_address,
```

Comment:
Probably won't be relevant with the refactoring above in mind, but `$current_address` is a reference to the current civi address; MURs should show the address we're trying to import (or both side by side in other scenarios).